BlueprintUI/Sources/Element/ElementContent.swift

```diff
@@ -23,42 +23,94 @@ public struct ElementContent {
         storage = builder
     }
 
-    // MARK: Measurement & Children
+    public var childCount: Int {
+        storage.childCount
+    }
+
+    fileprivate init(storage: ContentStorage) {
+        self.storage = storage
+    }
+
+    func measure(
+        in constraint: SizeConstraint,
+        with environment: Environment,
+        state: ElementState
+    ) -> CGSize {
+        autoreleasepool {
+            self.storage.measure(in: constraint, with: environment, state: state)
+        }
+    }
+
+    func performLayout(
+        in size: CGSize,
+        with environment: Environment,
+        state: ElementState
+    ) -> [LayoutResultNode] {
+        autoreleasepool {
+            storage.performLayout(
+                in: size,
+                with: environment,
+                state: state
+            )
+        }
+    }
+
+    func forEachElement(
+        in size: CGSize,
+        with environment: Environment,
+        children childNodes: [LayoutResultNode],
+        state: ElementState,
+        forEach: (ElementState, Element, LayoutResultNode) -> Void
+    ) {
+        autoreleasepool {
+            storage.forEachElement(
+                in: size,
+                with: environment,
+                children: childNodes,
+                state: state,
+                forEach: forEach
+            )
+        }
+    }
+}
+
+extension ElementContent {
 
     /// Measures the required size of this element's content.
     /// - Parameters:
     ///   - constraint: The size constraint.
     ///   - environment: The environment to measure in.
     /// - returns: The layout size needed by this content.
     public func measure(in constraint: SizeConstraint, environment: Environment) -> CGSize {
-        measure(
-            in: constraint,
-            environment: environment,
-            cache: CacheFactory.makeCache(name: "ElementContent")
-        )
-    }
+        autoreleasepool {
+            let element = MeasurementElement(content: self)
+            let root = ElementStateTree(name: "ElementContent.measure")
+
+            let (_, size) = root.performUpdate(with: element, in: environment) { state in
```

Comment:
Given `performUpdate` calls `finishedLayout`, we now call that on every measurement. It's a recursive call that has to have some overhead.

Reply:
This is for the "detached" measurement method; not for the regular measurement tree – eg where you're just invoking `element.content.measure` somewhere.